portfolio/src/main/webapp/comments.js

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Represents a Comments object whose getComments() method retrieves comments data from
+ * the Java servlet and populates the data into a container in the DOM. 
+ */
+class Comments {
+  constructor() {
+    this.commentsListContainer = document.getElementById("comments-container");
+    this.endpointToRetrieveDataFrom = "/comments-data";
+  }
+
+  /**
+   * Fetches a list of comments from a URL via the Java servlet, uses a promise 
+   * to add them to the DOM once the comments have been received.
+   */
+  getComments() {
+    fetch(this.endpointToRetrieveDataFrom)
+      .then(response => response.json())
+      .then((commentsList) => {
+        // Reset the commentsListContainer to reload it with the new list of comments.
+        this.commentsListContainer.innerHTML = '';
+
+        // Add each comment in the JSON to the DOM.
+        commentsList.forEach((comment) => {
+          this.commentsListContainer.appendChild(
+            this.createCommentElement(comment)
+          );
+        });
+      })
+      .catch(() => {
+        console.error("Failed to load comments.");
```

Comment:
In general, it's best practice to have descriptive error messages which make them much easier to debug. If an error occurred during fetch, you should report the status and response text. If an error occurred during json parsing, that's what your error should say.